tests/php/ORM/MySQLiConnectorTest.php

```diff
@@ -2,32 +2,48 @@
 
 namespace SilverStripe\ORM\Tests;
 
+use mysqli_driver;
 use SilverStripe\Dev\SapphireTest;
 use SilverStripe\Dev\TestOnly;
-use SilverStripe\ORM\Tests\MySQLiConnectorTest\MySQLiConnector;
+use SilverStripe\ORM\Connect\DatabaseException;
 use SilverStripe\ORM\DB;
+use SilverStripe\ORM\Tests\MySQLiConnectorTest\MySQLiConnector;
 use SilverStripe\Tests\ORM\Utf8\Utf8TestHelper;
 
 /**
  * @requires extension mysqli
  */
 class MySQLiConnectorTest extends SapphireTest implements TestOnly
 {
-    /**
-     * @dataProvider charsetProvider
-     */
-    public function testConnectionCharsetControl($charset, $defaultCollation)
+    private function getConnector(?string $charset = null, ?string $collation = null, bool $selectDb = false)
     {
         $config = DB::getConfig();
-        $config['charset'] = $charset;
-        $config['database'] = 'information_schema';
 
         if (strtolower(substr($config['type'] ?? '', 0, 5)) !== 'mysql') {
             return $this->markTestSkipped('The test only relevant for MySQL');
         }
 
+        if ($charset) {
```

Comment:
Moving the duplicate logic to this new private method is OK, except for the bit where the marking test as skipped if it's it's not running mysql - It makes it far less obvious could you please re-duplicate those bits and move those lines back to the individual tests